src/main/java/com/monov/cloud/gateway/dto/Course.java

```diff
@@ -0,0 +1,18 @@
+package com.monov.cloud.gateway.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Course {
+
+    private Long courseId;
```

Comment:
Whenever you have a model called for example Course, the id field does not need to be called courseId since it is implied that a field with name Id would be the id of the Course model. Look at your other models as well, they might have the same thing. 

Also I see that you are placing these models in a DTO package. If you intend them to be DTOs then name them CourseDTO etc. Since these DTO classes are going to be used in more than 1 microservice (their own microservice and the gateway) it might be good to move them outside into a common library/project to avoid duplicate classes.

Reply:
Fixes will be applied in DTOs branch's PR